prometheus-openstack-exporter

```diff
@@ -266,14 +268,16 @@ class DataGatherer(Thread):
         log.debug("Starting data gather thread")
         prodstack = {}
         while True:
+            log.debug("DataGatherer: Start refresh")
             start_time = time()
             try:
                 keystone, nova, neutron, cinder = get_clients()
 
+                log.debug("DataGatherer: Update keystone")
                 prodstack.update(self._get_keystone_info(keystone))
-                sleep(0)
```

Comment:
That is correct, these sleep(0) calls were added in https://github.com/canonical/prometheus-openstack-exporter/commit/546ece6cdd4fd9dadf8802392bbdae0fc8e2993b to try and yield intentionally more often, since the blocking versions of the libraries were being used which should have done it. It's not needed now.